components/base/wheeled/wheeled_base.go

```diff
@@ -115,6 +133,96 @@ func (base *wheeledBase) Spin(ctx context.Context, angleDeg, degsPerSec float64,
 	rpm, revolutions := base.spinMath(angleDeg, degsPerSec)
 
 	return base.runAll(ctx, -rpm, revolutions, rpm, revolutions)
+
+}
+
+func (base *wheeledBase) spinWithMovementSensor(ctx context.Context, angleDeg, degsPerSec float64, extra map[string]interface{}) error {
+	wheelrpm, revs := base.spinMath(angleDeg, degsPerSec)
+
+	currYaw, err := getCurrentYaw(ctx, base.orientationSensor, extra) // from -179 to 179
+	if err != nil {
+		return err
+	}
+
+	errCounter := 0
+	targetYaw := calculatedDomainLimitedAngleError(angleDeg, currYaw)
+	timer := time.NewTicker(time.Duration(pollTimeMs * float64(time.Millisecond))) //~rename
+	for {
+
+		select {
+		case <-ctx.Done():
+			return nil
+		default:
+		}
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-timer.C:
+		}
+		//~make into helper function and pass into timer case?
+		currYaw, err := getCurrentYaw(ctx, base.orientationSensor, extra) // from -179 to 179
+		if err != nil {
+			errCounter++
+			if errCounter > 100 {
+				break
+			}
+			return err
+		}
+
+		errAngle := targetYaw - currYaw
+		base.logger.Debugf("currYaw: %.2f, errAngle:%.2f", currYaw, errAngle)
+
+		// runAll calls GoFor, which has a necessary terminating condition of rotations reached
+		// poll the sensor for the current error in angle
+		// also check if we've overshot our target by fifteen degrees
+		if math.Abs(errAngle) < 5 || baseOvershot(currYaw, targetYaw, 15) {
+			if err := base.Stop(ctx, nil); err != nil {
+				return err
+			}
+			break
+		}
+		if err := base.runAll(ctx, -wheelrpm, revs, wheelrpm, revs); err != nil {
```

Comment:
[q] we do multiple run calls here, but the run calls are not incremental, right? If each runAll takes the total number of estimated revolutions necessary to get from the original angle to the target and we make that same call every 100 ms, aren't we actually increasing the probability that we overshoot? Should we not actually redo the math based on the current yaw at each iteration of the loop?

Reply:
I can rework it to call `runAll` once, have the IMU logic dictate the stop, and see how that works. Went back and forth because ideally (and as a future improvement) we would want a trajectory plan on this part of the code calling runAll (either on `GoFor` or `SetPower`) I don't want to go that far in this PR.

Reply:
keeping this open until I test again.

Reply:
Something is canceling context when I incrementally send in revolutions as in:
``` revIncrement := yallPollTime * rpm ```. I suspect it is `basic.GoFor`'s `opMgr.NewTimedWaitOp` but I don't want to go down that rabbit hole. Will make a ticket.